src/tests/userDetails.test.js

```diff
@@ -0,0 +1,298 @@
+import chai from 'chai';
+import { v4 as uuidv4 } from 'uuid';
+import chaiHttp from 'chai-http';
+import app from '../index.js';
+import UserDetailsModel from '../database/models/userDetails.model';
+import User from '../database/models/user.model.js';
+import { generateToken, hashPassword } from '../utils/index';
+import sequelize from '../database/config/db';
+import { userServices } from '../services';
+import redisClient from '../helpers/redis.js';
+import { isAuthenticated } from '../middleware/index.js';
+
+const { expect } = chai;
+chai.use(chaiHttp);
+
+const userInfo = {
+  email: 'testing@example.com',
+  password: 'Qwert@12345',
+};
+const testuser1 = {
+  id: '123',
+  displayName: 'John Doe',
+  email: 'johndoe@example.com',
+};
+describe('User Details Controller', function () {
+  let token;
+  let user;
+  before(async function () {
+    user = await userServices.getUserByEmail(userInfo.email);
+    const res = await chai.request(app).post('/users/login').send(userInfo);
+    token = res.body.token;
+  });
+
+  after(async function () {
+    await UserDetailsModel.destroy({ where: { userId: user.id } });
```

Comment:
Here you should use the services you would have created instead of calling the model.

Reply:
okay